Source/WTF/wtf/posix/ThreadingPOSIX.cpp

```diff
@@ -609,15 +609,17 @@ bool ThreadCondition::timedWait(Mutex& mutex, WallTime absoluteTime)
     if (absoluteTime < WallTime::now())
         return false;
 
-    if (absoluteTime > WallTime::fromRawSeconds(INT_MAX)) {
+    const time_t timetMax = sizeof(time_t) == 4 ? INT_MAX : INT64_MAX;
+
+    if (absoluteTime > WallTime::fromRawSeconds(timetMax)) {
```

Comment:
Is there any reason to not use `std::numeric_limits<time_t>` here?

The following should do:

```suggestion
    if (absoluteTime > WallTime::fromRawSeconds(std::numeric_limits<time_t>::max())) {
```

Reply:
The only reason is that my C++ knowledge is quite limited, will update the branch !

Reply:
Source of a new clang warning:
```
/app/webkit/Source/WTF/wtf/posix/ThreadingPOSIX.cpp:612:49: warning: implicit conversion from 'long' to 'double' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
    if (absoluteTime > WallTime::fromRawSeconds(std::numeric_limits<time_t>::max())) {
                       ~~~~~~~~                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```